Assignment 5.Rmd

```diff
@@ -40,42 +46,106 @@ We want to see if we can build a decision tree to help teachers decide which stu
+                    
 ```
 
 #Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
 ```{r}
-score_ctree <- 
+#outcome ~ predictor1+predictor2+predictor3+ect.
+
+score_ctree <- rpart(advice ~ prior_prob_count + prior_percent_correct +hints, method="class", data=D1)
+  
 ```
 
 #Plot tree
 ```{r}
 
+rpart.plot(score_ctree)
+#https://blog.exploratory.io/visualizing-a-decision-tree-using-r-packages-in-explortory-b26d4cb5e71f
+
+printcp(score_ctree)
+
+#plot(score_ctree)
+
 ```
 
 Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
 
+```{r}
+#Interpretation: Depending on whether they had hints over 20, the actions split into two (monitor for 7% and no action for 93%). It means that if a student received hints more or equal to 20 times, 7% chance that teacher shouldn monitor. And if they received hints over 58times, 2% chance that teacher should intervene and 8% should be moniotred. 
+#If they received hints less than 20, 93% chance that teacher should ignore. However, when students got less 20 hints, previous correct less then 60% and got more than 1 hints, 19% of them should be monitored.
+
+#how many hints students got and prior percent correct scores will be two behaviors that teachers should most closely pay attention to. (because they are in the higher hierarchy)
+
+
+```
+
+
 #Test Tree
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
 ```{r}
 #Upload new data
 
-D2 <- 
+D2 <- read.csv("intelligent_tutor_new.csv")
 
 #Generate predicted advice using the predict() command for new students based on tree generated from old students
 
-D2$prediction <- 
+#predict(old, new)
+D2$prediction <- predict(score_ctree, D2)
 
 ``` 
 ## Part III
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
 
+```{r}
+#advice action in D2 obs.data
+D2$advice <- with(D2, ifelse(score >= 0.3,
+                    ifelse(score >= 0.6,"no action","monitor"),
+                    "intervene"))
+#advice from the prediction
+D2$advice_Pred<-with(D2, ifelse(prediction >= 0.3,
+                    ifelse(prediction >= 0.6,"no action","monitor"),
+                    "intervene"))
+
+#compare prediciton action and action.By changing the cutoff score, we want the rel error be higher(it is how we prediction) and xerror (cross validation )be lower
+D2$Accuracy<- ifelse(D2$advice==D2$advice_Pred,1, 0)
+
+#calculate the accuracy of the prediction
+sum(as.numeric(D2$Accuracy))/200
+
+#prediction advice accuracy is 71.5%
+
+
+#0.3-0.6 71.5%
+#0.7-0.8 47%
+
```

Comment:
Overall, great job but try to provide an analysis in Part III. As in, state whether the model is generalizable or whether it overfits.